### DISCUSSION TITLE
What is the difference between --no-restart and --monitor-only?

### DISCUSSION THREAD

**Discussion by kwint** [containrrr/watchtower#1109] [Q&A] [answered]
What is the difference between --monitor-only and --no-restart?
What I get from the docs is that they both pull new images, but just don't restart the container?

Or does --monitor-only pull the images but doesn't keep them?

**Comment by piksel** [comment]
I actually have no idea. I tried looking at what they do in the code. And they are not aliases at least, but I don't really understand why both are there. The effect seems to be roughly the same...

**Reply by piksel** [reply]
I think no-restart might leave the container in a stopped state. Maybe that's useful?

**Comment by simonaronsson** [comment, marked as answer]
`--monitor-only` pulls the new image but leaves the container as-is. `--no-restart` recreates it, but keeps it turned off, mainly to allow for state orchestration through some other means, like systemd.
